### PR TITLE
add instance refresh preferences

### PR DESCRIPTION
**Pull request by scott-doyland-burrows**
## what

Adds ```skip_matching``` and ```auto_rollback``` to the instance refresh preferences.

Allows the instance refresh preferences to be optional.  If the values are not set in the calling module, then the default values are used as per:

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/autoscaling_group#instance_refresh

## why

Previously it was not possible to set the ```skip_matching``` and ```auto_rollback```.

It was also not possible to omit preferences from the calling module and just use the defaults instead.

With this change we can now set preferences and/or omit preferences that we wish to use the defaults for.

The change should not cause any diffs in current infrastructure.

## references

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/autoscaling_group#instance_refresh

## examples

It is now possible to use something such as this to use defaults:

```
  instance_refresh = ({
    strategy = "Rolling"
  })
```

or this to override certain defaults:

```
  instance_refresh = ({
    strategy = "Rolling"
    preferences = ({
      min_healthy_percentage = 50
      skip_matching          = true
    })
    triggers = ["tags"]
  })
```

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -112,10 +112,12 @@ variable "instance_refresh" {
   type = object({
     strategy = string
     preferences = object({
```
Comment (goruha) [inline variables.tf:114]:
```suggestion
    preferences = optional(object({
```

**Review by goruha — DISMISSED** [review]
@scott-doyland-burrows How about making `preferences`  also optional?

File: variables.tf (hunk 1 above)
Comment (scott-doyland-burrows) [reply variables.tf:114]: That's fine by me.  I will give it a quick test tomorrow and then make the change.

Comment (scott-doyland-burrows) [reply variables.tf]: I have made the required changes.  Please can you review them.

**Comment by Gowiem** [comment]
@scott-doyland-burrows this looks good and we'd be happy to get it merged. Can you please do the following and push the changes?

```
make init
make readme
```

Thanks!

**Comment by scott-doyland-burrows** [comment]
I have run ```make init``` and ```make readme``` from the root of the repo and pushed the chanes.

**Comment by Gowiem** [comment]
Ah @scott-doyland-burrows -- can you update https://github.com/cloudposse/terraform-aws-ec2-autoscale-group/blob/main/versions.tf#L2 to require 1.3+? Otherwise someone could have 1.0 or 1.1 pinned, try to use this module, and then it'll break on `optional`.

**Comment by scott-doyland-burrows** [comment]
> Ah @scott-doyland-burrows -- can you update https://github.com/cloudposse/terraform-aws-ec2-autoscale-group/blob/main/versions.tf#L2 to require 1.3+? Otherwise someone could have 1.0 or 1.1 pinned, try to use this module, and then it'll break on `optional`.

I have made the change.

**Comment by Gowiem** [comment]
Ah apologies @scott-doyland-burrows -- Can you run the `make readme` again and commit / push the results? The change to the required TF version causes changes in our README.

**Comment by scott-doyland-burrows** [comment]
> Ah apologies @scott-doyland-burrows -- Can you run the `make readme` again and commit / push the results? The change to the required TF version causes changes in our README.

I have run:
```make init``` and ```make readme```

**Review by Gowiem — APPROVED** [review]

**Pull request merged**